### PR TITLE
Fix/setup

### PR DESCRIPTION
**Pull request by ntarocco**
From: https://github.com/inveniosoftware/idutils/pull/109

### CODE REVIEW

File: idutils/ext.py [hunk 1]
```diff
@@ -51,6 +51,8 @@ def get_scheme_config_func():
 from threading import Lock
 
 from importlib_metadata import entry_points
+# when Python >=3.12, remove importlib_metadata and replace with:
+# from importlib.metadata import entry_points
```
Comment (zzacharo) [inline idutils/ext.py:55]: minor: I guess we don't want a python check and already import the correct package there?

**Review by zzacharo — APPROVED** [review]

File: idutils/ext.py (hunk 1 above)
Comment (ntarocco) [reply idutils/ext.py:55]: I am hesitating, because this will be copy/pasted around for sure.... not sure...

**Comment by ntarocco** [comment]
I am fixing the tests

File: setup.cfg [hunk 2]
```diff
@@ -32,15 +32,15 @@ packages = find:
...
 install_requires =
-    six>=1.10
+    importlib-metadata>=5.0  # remove when Python >=3.12
     isbnlib>=3.10.8
 
 [options.extras_require]
 tests =
     pytest-black-ng>=0.4.0
     pytest-cache>=1.0
+    pytest-invenio>=1.4.0  # required for fixtures
```
Comment (ntarocco) [inline setup.cfg:42]: @tmorrell I had to put it back, @zzacharo very wisely suggested that it was necessary for the [entry_points](https://github.com/inveniosoftware/pytest-invenio/blob/d03d0b541a4e74c57a0437150a34329ae0716246/pytest_invenio/fixtures.py#L810) fixtures (and there is a lot code there).

Comment (tmorrell) [reply setup.cfg:42]: Yup, the thought was we could pull out the entry point code....but probably not worth the effort.

File: idutils/ext.py [hunk 3]
```diff
@@ -54,6 +54,9 @@ def get_scheme_config_func():
 
 from .schemes import IDUTILS_PID_SCHEMES
 
+# when Python >=3.12, remove importlib_metadata and replace with:
+# from importlib.metadata import entry_points
```
Comment (tmorrell) [inline idutils/ext.py:58]: Have you tried that in 3.12? I don't think it worked for me.

**Review by tmorrell — APPROVED** [review]
Looks great, Thanks!

File: idutils/ext.py (hunk 3 above)
Comment (ntarocco) [reply idutils/ext.py:58]: Tests are passing, and it should work because importlib_metadata >= 5 has the same implementation as Python 3.12.

**Pull request merged**